### ISSUE TITLE
[Bug]: mo hung in Multi-CN e2e BVT(Race) Test

### ISSUE THREAD

**Issue by guguducken** [matrixorigin/matrixone#12881]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 5fd25c0601621808678b8ece0e76036f6b6cf470
- Hardware parameters:
- OS type:
- Others: https://github.com/matrixorigin/matrixone/actions/runs/6952738419/job/18916785790
```

### Actual Behavior

<img width="1423" alt="image" src="https://github.com/matrixorigin/matrixone/assets/22561920/5076fa7c-4f5d-4afb-a2eb-f6211ffb840e">
more detail: https://github.com/matrixorigin/matrixone/actions/runs/6952738419/job/18916785790

**Comment by arjunsk** [comment]
I think this is related to https://github.com/matrixorigin/matrixone/issues/12166

@volgariver6 Could you kindly look into this or assign it to the appropriate person?

**Comment by volgariver6** [comment]
multi cn issue, related to https://github.com/matrixorigin/matrixone/issues/12544, @zhangxu19830126 is handling it.

**Comment by zhangxu19830126** [comment]
maybe fixed by #13001
